### PR TITLE
Clear the Set-Cookie header before appending all the cookies, fixes #339

### LINKED ISSUE #339: Set-Cookie header contains duplicate cookies when adding more than one value
It seems that every time a cookie value is added to `response.http.cookies`, all of the cookies are appended to the `Set-Cookie` response header.

So if I have `cookies[a] = 1`, then do `cookies[b] = 2`, the resulting header is `Set-Cookie: a,a,b`

If I then do `cookies[c] = 3`, the result is `Set-Cookie: a,a,b,a,b,c` and so on.

My specific case is a Vapor 3 app using `SessionsMiddleware`, I added another middleware `CSRFMiddleware`:

```swift
import Vapor

final class CSRFMiddleware: Middleware, ServiceType {
    static func makeService(for container: Container) throws -> Self {
        return .init()
    }
    
    func respond(to req: Request, chainingTo next: Responder) throws -> EventLoopFuture<Response> {
        return try next.respond(to: req).map { res in
            res.http.cookies["csrf"] = HTTPCookieValue(string: "nope", expires: nil, maxAge: nil, domain: nil, path: nil, isSecure: false, isHTTPOnly: false, sameSite: nil)
            return res
        }
    }
}
```

In `configure.swift`:
```swift
services.register(CSRFMiddleware.self)
...
middlewares.use(CSRFMiddleware.self)
middlewares.use(SessionsMiddleware.self)
```

Let me know if this is an invalid setup or there's something else I'm missing.

### PR DESCRIPTION
**Pull request by joewalsh**
Fix for #339

### CODE REVIEW

**Comment by joewalsh** [comment]
@tanner0101 thanks for the feedback! I added a test and I am willing to re-submit a PR against the `3` branch after this is approved.

File: Tests/HTTPKitTests/HTTPTests.swift [hunk 1]
```diff
@@ -33,6 +33,19 @@ class HTTPTests: HTTPKitTestCase {
         
         XCTAssertEqual(httpReq.headers.firstValue(name: .cookie), "id=value")
     }
+    
+    func testMultipleCookiesAreSerializedCorrectly() throws {
```
Comment (tanner0101) [inline Tests/HTTPKitTests/HTTPTests.swift]: This test needs to be added to the `allTests` array here: https://github.com/vapor/http/blob/master/Tests/LinuxMain.swift

**Comment by tanner0101** [comment]
Looks like there were some breaking changes in NIO 2 that NIOHPACK needs to be updated for. Once @Lukasa updates that we can re-test and merge.

File: Tests/HTTPKitTests/HTTPTests.swift (hunk 1 above)
Comment (joewalsh) [reply Tests/HTTPKitTests/HTTPTests.swift]: Fixed

**Comment by penny-coin** [comment]
Hey @joewalsh, you just merged a pull request, have a coin! 

You now have 1 coins.

**Pull request merged**